test/Common.FunctionalTests/Utilities/Helpers.cs

```diff
@@ -73,5 +74,64 @@ public static void CopyFiles(DirectoryInfo source, DirectoryInfo target, ILogger
                 fileInfo.CopyTo(destFileName);
             }
         }
+
+        public static void ModifyWebConfig(this DeploymentResult deploymentResult, Action<XElement> action)
+        {
+            var webConfigPath = Path.Combine(deploymentResult.ContentRoot, "web.config");
+            var document = XDocument.Load(webConfigPath);
+
+            document.Save(webConfigPath);
+        }
+
+        public static Task<HttpResponseMessage> RetryRequestAsync(this HttpClient client, string uri, Func<HttpResponseMessage, bool> predicate)
+        {
+            return RetryRequestAsync(client, uri, message => Task.FromResult(predicate(message)));
+        }
+
+        public static async Task<HttpResponseMessage> RetryRequestAsync(this HttpClient client, string uri, Func<HttpResponseMessage, Task<bool>> predicate)
+        {
+            HttpResponseMessage response = await client.GetAsync(uri);
+
+            for (var i = 0; i < RetryRequestCount && !await predicate(response); i++)
+            {
+                // Keep retrying until app_offline is present.
+                response = await client.GetAsync(uri);
+                await Task.Delay(RetryRequestDelay);
+            }
+
+            if (!await predicate(response))
+            {
+                throw new InvalidOperationException($"Didn't get response that satisfies predicate after {RetryRequestCount} retries");
+            }
+
+            return response;
+        }
+
+        public static void AssertWorkerProcessStop(this IISDeploymentResult deploymentResult)
+        {
+            var hostProcess = deploymentResult.HostProcess;
+            Assert.True(hostProcess.WaitForExit((int)TimeoutExtensions.DefaultTimeout.TotalMilliseconds));
+
+            if (deploymentResult.DeploymentParameters.ServerType == ServerType.IISExpress)
+            {
+                Assert.Equal(0, hostProcess.ExitCode);
```

Comment:
Why can't this check be done for w3wp?

Reply:
When you don't start process you can't observe it's exit code.